src/PowerShell/AzPSLogin.ts

```diff
@@ -0,0 +1,93 @@
+    }
+
+    async login() {
+        core.info(`Running Azure PowerShell Login.`);
+        this.setPSModulePathForGitHubRunner();
+        await this.setPSModulePathForLatestAzAccounts();
+
+        const [loginMethod, loginScript] = await AzPSScriptBuilder.getAzPSLoginScript(this.loginConfig);
+        core.info(`Attempting Azure PowerShell login by using ${loginMethod}...`);
+        core.debug(`Azure PowerShell Login Script: ${loginScript}`);
+        await AzPSLogin.runPSScript(loginScript);
+        console.log(`Running Azure PowerShell Login successfully.`);
+    }
+
+    setPSModulePathForGitHubRunner() {
+        const runner: string = process.env.RUNNER_OS || os.type();
+        switch (runner.toLowerCase()) {
+            case "linux":
+                this.pushPSModulePath(AzPSConstants.DEFAULT_AZ_PATH_ON_LINUX);
+                break;
+            case "windows":
+            case "windows_nt":
+                this.pushPSModulePath(AzPSConstants.DEFAULT_AZ_PATH_ON_WINDOWS);
+                break;
+            case "macos":
+            case "darwin":
+                core.warning(`Skip setting the default PowerShell module path for OS ${runner.toLowerCase()}.`);
+                break;
+            default:
+                core.warning(`Skip setting the default PowerShell module path for unknown OS ${runner.toLowerCase()}.`);
+                break;
+        }
+    }
+
+    async setPSModulePathForLatestAzAccounts() {
+        let getLatestAccountsScript: string = AzPSScriptBuilder.getLatestModulePathScript(AzPSConstants.AzAccounts);
+        core.debug(`The script to get the latest Az.Accounts path: ${getLatestAccountsScript}`);
+        let azAccountsLatestPath: string = await AzPSLogin.runPSScript(getLatestAccountsScript);
+        core.debug(`The latest Az.Accounts path used: ${azAccountsLatestPath}`);
+        this.pushPSModulePath(azAccountsLatestPath);
+    }
+
+    pushPSModulePath(psModulePath: string) {
+        process.env.PSModulePath = `${psModulePath}${path.delimiter}${process.env.PSModulePath}`;
+        core.debug(`Set PSModulePath as ${process.env.PSModulePath}`);
+    }
+
+    static async runPSScript(psScript: string): Promise<string> {
+        let outputString: string = "";
+        let commandStdErr = false;
+        const options: any = {
+            silent: true,
+            listeners: {
+                stdout: (data: Buffer) => {
+                    outputString += data.toString();
+                },
+                stderr: (data: Buffer) => {
+                    let error = data.toString();
+                    if (error && error.trim().length !== 0) {
+                        commandStdErr = true;
+                        core.error(error);
+                    }
+                }
+            }
+        };
+
+        let psPath:string = await io.which(AzPSConstants.PowerShell_CmdName, true);
+        await exec.exec(`"${psPath}" -Command`, [psScript], options)
```

Comment:
```suggestion
        await exec.exec(`"${psPath}"`, [' -Command', psScript], options)
```
Let's move the argument `-Command` to `args`.

Reply:
Done.